test/module/irohad/synchronizer/synchronizer_test.cpp

```diff
@@ -200,7 +216,8 @@ TEST_F(SynchronizerTest, ExactlyThreeRetrievals) {
   DefaultValue<expected::Result<std::unique_ptr<MutableStorage>, std::string>>::
       SetFactory(&createMockMutableStorage);
   EXPECT_CALL(*mutable_factory, createMutableStorage()).Times(1);
-  EXPECT_CALL(*mutable_factory, commit_(_)).Times(1);
+  EXPECT_CALL(*mutable_factory, commit_(_))
+      .WillOnce(Return(ByMove(std::unique_ptr<LedgerState>{})));
```

Comment:
Returning `nullptr` does not seem like a good idea, maybe reuse `ledger_peers` and assertions? Same below.